core.go

```diff
@@ -412,6 +418,49 @@ func (m *Mat) SetDoubleAt3(x, y, z int, val float64) {
 	C.Mat_SetDouble3(m.p, C.int(x), C.int(y), C.int(z), C.double(val))
 }
 
+// ToImage returns Go image.Image from Mat.
+func (m *Mat) ToImage() (image.Image, error) {
+	t := m.Type()
+	if t != int(MatTypeCV8U) && t != int(MatTypeCV8UC1) && t != int(MatTypeCV8UC3) && t != int(MatTypeCV8UC4) {
+		return nil, errors.New("ToImage supports only MatType CV8U, CV8UC1, CV8UC3 and CV8UC4")
+	}
+
+	var height, width, channels, step int = m.Rows(), m.Cols(), m.Channels(), m.Step()
+
+	data := m.ToBytes()
+
+	if t == int(MatTypeCV8U) || t == int(MatTypeCV8UC1) {
+		out := image.NewGray(image.Rect(0, 0, width, height))
+		c := color.Gray{Y: uint8(0)}
+
+		for y := 0; y < height; y++ {
+			for x := 0; x < width; x++ {
+				c.Y = uint8(data[y*step+x])
+				out.SetGray(x, y, c)
+			}
+		}
+
+		return out, nil
+	} else {
```

Comment:
No need for a `else` here, you return in the previous `if`.

Reply:
Fixed.